### PR TITLE
license, openGraph tests: convert to testify

### LINKED ISSUE #12860: Migrate tests from "api4/license_test.go" and "api4/openGraph_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `api4/license_test.go` and `api4/openGraph_test.go` files.

The expected way to migrate this is to go to the `api4/license_test.go` and `api4/openGraph_test.go` files and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in [https://community.mattermost.com/](https://community.mattermost.com/)

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19625

### PR DESCRIPTION
**Pull request by drekar**
#### Summary
Update `api4/license_test.go` and `api4/openGraph_test.go` to use `testify`.

#### Ticket Link
Fixes #12860 
[MM-19625](https://mattermost.atlassian.net/browse/MM-19625)

### CODE REVIEW

File: api4/license_test.go [hunk 1]
```diff
@@ -15,22 +17,22 @@ func TestGetOldClientLicense(t *testing.T) {
 	license, resp := Client.GetOldClientLicense("")
 	CheckNoError(t, resp)
 
-	if len(license["IsLicensed"]) == 0 {
-		t.Fatal("license not returned correctly")
-	}
+	require.NotEqual(t, license["IsLicensed"], 0, "license not returned correctly")
```
Comment (hanzei) [inline api4/license_test.go]: This should be
```suggestion
	require.NotEqual(t, license["IsLicensed"], "", "license not returned correctly")
```

File: api4/license_test.go [hunk 2]
```diff
@@ -15,22 +17,22 @@ func TestGetOldClientLicense(t *testing.T) {
...
 	_, resp = Client.GetOldClientLicense("")
 	CheckNoError(t, resp)
 
-	if _, err := Client.DoApiGet("/license/client", ""); err == nil || err.StatusCode != http.StatusNotImplemented {
-		t.Fatal("should have errored with 501")
-	}
+	_, err := Client.DoApiGet("/license/client", "")
+	require.Error(t, err, "get /license/client did not return an error")
+	require.Equalf(t, err.StatusCode, http.StatusNotImplemented,
+		"expected 501 Not Implemented, got %v", err.StatusCode)
```
Comment (hanzei) [inline api4/license_test.go]: You don't need to log the status code. testify will do this for you.

File: api4/openGraph_test.go [hunk 3]
```diff
@@ -61,19 +61,14 @@ func TestGetOpenGraphMetadata(t *testing.T) {
...
-		}
 
-		if ogDataCacheMissCount != data["cacheMissCount"].(int) {
-			t.Fatal(fmt.Sprintf(
-				"Cache miss count didn't match. Expected value %d. Actual value %d.",
-				data["cacheMissCount"].(int), ogDataCacheMissCount,
-			))
-		}
+		require.Equalf(t, openGraph["title"], data["title"].(string),
+			"OG data title mismatch for path \"%s\". Expected title: \"%s\". Actual title: \"%s\"",
```
Comment (hanzei) [inline api4/openGraph_test.go]: Same here. You don't need to log them.

File: api4/license_test.go (hunk 2 above)
Comment (drekar) [reply api4/license_test.go]: Good to know, thanks; fixed.

File: api4/license_test.go (hunk 1 above)
Comment (drekar) [reply api4/license_test.go]: Ah yes, you are absolutely right; fixed.

File: api4/openGraph_test.go (hunk 3 above)
Comment (hanzei) [reply api4/openGraph_test.go]: It seams like you did not fix this

Comment (drekar) [reply api4/openGraph_test.go]: This one is now updated in the same way as those in `license_test` to rely on the verbosity of `testify`; thanks!

File: api4/license_test.go [hunk 4]
```diff
@@ -4,6 +4,8 @@ import (
 	"net/http"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
```
Comment (hanzei) [inline api4/license_test.go]: Please remove this line

File: api4/license_test.go [hunk 5]
```diff
@@ -15,22 +17,22 @@ func TestGetOldClientLicense(t *testing.T) {
...
 
 	_, resp = Client.GetOldClientLicense("")
 	CheckNoError(t, resp)
 
-	if _, err := Client.DoApiGet("/license/client", ""); err == nil || err.StatusCode != http.StatusNotImplemented {
-		t.Fatal("should have errored with 501")
-	}
+	_, err := Client.DoApiGet("/license/client", "")
+	require.Error(t, err, "get /license/client did not return an error")
+	require.Equal(t, err.StatusCode, http.StatusNotImplemented,
```
Comment (hanzei) [inline api4/license_test.go]: Nit: you can combine these lines.

File: api4/license_test.go [hunk 6]
```diff
@@ -15,22 +17,22 @@ func TestGetOldClientLicense(t *testing.T) {
...
+		"expected 501 Not Implemented")
 
-	if _, err := Client.DoApiGet("/license/client?format=junk", ""); err == nil || err.StatusCode != http.StatusBadRequest {
-		t.Fatal("should have errored with 400")
-	}
+	_, err = Client.DoApiGet("/license/client?format=junk", "")
+	require.Error(t, err, "get /license/client?format=junk did not return an error")
+	require.Equal(t, err.StatusCode, http.StatusBadRequest,
```
Comment (hanzei) [inline api4/license_test.go]: Nit: you can combine these lines.

**Comment by drekar** [comment]
Rebasing branch on latest upstream master (caused 4 removed commits, so force pushed to keep the merge log clean)

**Pull request merged**